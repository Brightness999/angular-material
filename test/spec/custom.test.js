/**
 * @fileOverview Test logging of custom data objects.
 */
const Logality = require('../..');
const { sink, stubLogality } = require('../lib/tester.lib');

const DATA = {
  id: 10,
  email: 'one@go.com',
};

describe('Custom Serializer', () => {
  stubLogality();

  test('Will log custom data properly', (done) => {
    const logality = new Logality({
      appName: 'testLogality',
      wstream: sink((chunk) => {
        expect(chunk).toMatchSnapshot();
        done();
      }),
    });

    const log = logality.get();

    log('info', 'hello world', { custom: DATA });
  });
});
