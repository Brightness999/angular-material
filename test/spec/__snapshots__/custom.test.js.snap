// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Serializer Will log custom data properly 1`] = `
Object {
  "context": Object {
    "custom": Object {
      "email": "one@go.com",
      "id": 10,
    },
    "runtime": Object {
      "application": "testLogality",
    },
    "source": Object {
      "file_name": "/test/spec/custom.test.js",
    },
    "system": Object {
      "hostname": "localhost",
      "pid": 36255,
      "process_name": "node .",
    },
  },
  "dt": "2018-05-18T16:25:57.815Z",
  "event": Object {},
  "level": "info",
  "message": "hello world",
  "severity": 6,
}
`;
